src/flow/kishimoto.jl

```diff
@@ -0,0 +1,75 @@
+# Method when using Boykov-Kolmogorov as a subroutine
+# Kishimoto algorithm
+function kishimoto{T<:AbstractFloat}(
+  flow_graph::DiGraph,                       # the input graph
+  source::Int,                               # the source vertex
+  target::Int,                               # the target vertex
+  capacity_matrix::AbstractArray{T,2},       # edge flow capacities
+  flow_algorithm::BoykovKolmogorovAlgorithm, # keyword argument for algorithm
+  routes::Int                                # keyword argument for routes
+  )
+  # Initialisation
+  flow, F, labels = maximum_flow(flow_graph, source, target,
+         capacity_matrix, algorithm = flow_algorithm)
+  restriction = flow / routes
+  flow, F, labels = maximum_flow(flow_graph, source, target, capacity_matrix,
+         algorithm = flow_algorithm, restriction = restriction)
+
+  # Loop
+  i = 1
+  while flow ≉ routes * restriction && flow < routes * restriction
```

Comment:
I'll leave it to you, but are you **sure** that FP precision will change things like this (such that `a != b` will evaluate to `true` even if `a` is technically equal to `b`)?